server/routes/upgradeMember.ts

```diff
@@ -5,41 +5,42 @@ import { factory, formatFilename } from '../util/logging';
 const log = factory.getLogger(formatFilename(__filename));
 
 const upgradeMember = async (models, req: UserRequest, res: Response, next: NextFunction) => {
+  const { Op } = models.sequelize;
   const [chain, community] = await lookupCommunityIsVisibleToUser(models, req.body, req.user, next);
   const { address, new_role } = req.body;
   if (!address) return next(new Error('Invalid Address'));
   if (!new_role) return next(new Error('Invalid Role'));
   if (!req.user) return next(new Error('Not logged in'));
   // if chain is present we know we are dealing with a chain first community
   const chainOrCommObj = (chain) ? { chain_id: chain.id } : { offchain_community_id: community.id };
+  const requesterAddresses = await req.user.getAddresses();
+  const requesterAddressIds = Array.from(requesterAddresses.map((a) => a.id));
```

Comment:
does the map alone not get you an array that works?

Reply:
the query need a specific value, the `address_id`. maybe I'm missing something, but i couldn't get the query to work with passing in the Address object array and only looking at a specific property, `id`. the second array is just of the ids so that it could be passed in to `[op.in]` without hitch.

Reply:
oh, you meant `Array.from(...)`. Map should work, but in the case that `requesterAddresses` is not an array, `Array.from` forces the empty/null/error'd return of the map function to be an empty array, which will be caught not as a type-error/logic error, but in the query+subsequent "no role/not admin" checks.

Reply:
It doesn't really need to be here, upon inquiry into the Array.from and map documentation.